<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>HttpppMainWindow</class>
 <widget class="QMainWindow" name="HttpppMainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>757</width>
    <height>564</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../icons.qrc">
    <normaloff>:/icons/httppp.svg</normaloff>:/icons/httppp.svg</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QWidget" name="panelControls" native="true">
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <property name="margin">
        <number>0</number>
       </property>
       <item>
        <widget class="QPushButton" name="pushLoad">
         <property name="text">
          <string>Load File</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushReload">
         <property name="text">
          <string>Reload</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushHitsToCSV">
         <property name="text">
          <string>HTTP Hits to CSV</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="pushTcpConversations">
         <property name="text">
          <string>TCP Conversations</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushTcpPackets">
         <property name="text">
          <string>TCP Packets</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushHttpHits">
         <property name="text">
          <string>HTTP Hits</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushDetails">
         <property name="text">
          <string>Details</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushMessages">
         <property name="text">
          <string>Messages</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="panelPanels" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <property name="margin">
        <number>0</number>
       </property>
       <item>
        <widget class="QGroupBox" name="panelTcpConversations">
         <property name="title">
          <string>TCP Conversations</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <item>
           <widget class="EnhancedTableView" name="tcpConversationsView">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="alternatingRowColors">
             <bool>true</bool>
            </property>
            <property name="selectionMode">
             <enum>QAbstractItemView::SingleSelection</enum>
            </property>
            <property name="selectionBehavior">
             <enum>QAbstractItemView::SelectRows</enum>
            </property>
            <property name="gridStyle">
             <enum>Qt::DotLine</enum>
            </property>
            <property name="sortingEnabled">
             <bool>true</bool>
            </property>
            <attribute name="verticalHeaderVisible">
             <bool>false</bool>
            </attribute>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="panelTcpPackets">
         <property name="title">
          <string>TCP Packets</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <widget class="EnhancedTreeView" name="tcpPacketsView">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="panelHttpHits">
         <property name="title">
          <string>HTTP Hits</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QGroupBox" name="groupBox">
            <property name="title">
             <string>Custom Fields</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_2">
             <item row="1" column="0">
              <widget class="QComboBox" name="comboField1">
               <property name="editable">
                <bool>true</bool>
               </property>
               <property name="insertPolicy">
                <enum>QComboBox::InsertAtTop</enum>
               </property>
               <property name="frame">
                <bool>false</bool>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QComboBox" name="comboField2">
               <property name="editable">
                <bool>true</bool>
               </property>
               <property name="insertPolicy">
                <enum>QComboBox::InsertAtTop</enum>
               </property>
               <property name="sizeAdjustPolicy">
                <enum>QComboBox::AdjustToContentsOnFirstShow</enum>
               </property>
               <property name="frame">
                <bool>false</bool>
               </property>
              </widget>
             </item>
             <item row="1" column="2">
              <widget class="QComboBox" name="comboField3">
               <property name="editable">
                <bool>true</bool>
               </property>
               <property name="insertPolicy">
                <enum>QComboBox::InsertAtTop</enum>
               </property>
               <property name="frame">
                <bool>false</bool>
               </property>
              </widget>
             </item>
             <item row="0" column="0">
              <widget class="QWidget" name="widget_2" native="true">
               <layout class="QHBoxLayout" name="horizontalLayout_3">
                <property name="margin">
                 <number>0</number>
                </property>
                <item>
                 <widget class="MultiPixmapButton" name="switchField1">
                  <property name="minimumSize">
                   <size>
                    <width>16</width>
                    <height>16</height>
                   </size>
                  </property>
                  <property name="maximumSize">
                   <size>
                    <width>16</width>
                    <height>16</height>
                   </size>
                  </property>
                  <property name="toolTip">
                   <string>Click to select packet direction</string>
                  </property>
                  <property name="scaledContents">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLabel" name="label">
                  <property name="text">
                   <string>Field 1</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QWidget" name="widget_3" native="true">
               <layout class="QHBoxLayout" name="horizontalLayout_4">
                <property name="margin">
                 <number>0</number>
                </property>
                <item>
                 <widget class="MultiPixmapButton" name="switchField2">
                  <property name="minimumSize">
                   <size>
                    <width>16</width>
                    <height>16</height>
                   </size>
                  </property>
                  <property name="maximumSize">
                   <size>
                    <width>16</width>
                    <height>16</height>
                   </size>
                  </property>
                  <property name="toolTip">
                   <string>Click to select packet direction</string>
                  </property>
                  <property name="scaledContents">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLabel" name="label_2">
                  <property name="text">
                   <string>Field 2</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
             <item row="0" column="2">
              <widget class="QWidget" name="widget_4" native="true">
               <layout class="QHBoxLayout" name="horizontalLayout_5">
                <property name="margin">
                 <number>0</number>
                </property>
                <item>
                 <widget class="MultiPixmapButton" name="switchField3">
                  <property name="minimumSize">
                   <size>
                    <width>16</width>
                    <height>16</height>
                   </size>
                  </property>
                  <property name="maximumSize">
                   <size>
                    <width>16</width>
                    <height>16</height>
                   </size>
                  </property>
                  <property name="toolTip">
                   <string>Click to select packet direction</string>
                  </property>
                  <property name="scaledContents">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLabel" name="label_3">
                  <property name="text">
                   <string>Field 3</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="EnhancedTableView" name="httpHitsView">
            <property name="alternatingRowColors">
             <bool>true</bool>
            </property>
            <property name="selectionMode">
             <enum>QAbstractItemView::SingleSelection</enum>
            </property>
            <property name="selectionBehavior">
             <enum>QAbstractItemView::SelectRows</enum>
            </property>
            <property name="gridStyle">
             <enum>Qt::DotLine</enum>
            </property>
            <property name="sortingEnabled">
             <bool>true</bool>
            </property>
            <attribute name="verticalHeaderVisible">
             <bool>false</bool>
            </attribute>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="panelDetails">
         <property name="title">
          <string>Details</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QTextBrowser" name="detailsView">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="panelMessages">
      <property name="title">
       <string>Messages</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_6">
       <item>
        <widget class="QPlainTextEdit" name="messages"/>
       </item>
       <item>
        <widget class="QWidget" name="widget" native="true">
         <layout class="QVBoxLayout" name="verticalLayout_6">
          <property name="margin">
           <number>0</number>
          </property>
          <item>
           <widget class="QPushButton" name="pushCloseMessages">
            <property name="text">
             <string>Close</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushClearMessages">
            <property name="text">
             <string>Clear</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushEnableMessages">
            <property name="text">
             <string>Enable</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushAutoOpenMessages">
            <property name="text">
             <string>Auto Open</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>102</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>757</width>
     <height>18</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>EnhancedTreeView</class>
   <extends>QTreeView</extends>
   <header>widget/enhancedtreeview.h</header>
   <slots>
    <signal>currentIndexChanged(QModelIndex)</signal>
   </slots>
  </customwidget>
  <customwidget>
   <class>EnhancedTableView</class>
   <extends>QTableView</extends>
   <header>widget/enhancedtableview.h</header>
   <slots>
    <signal>currentIndexChanged(QModelIndex)</signal>
   </slots>
  </customwidget>
  <customwidget>
   <class>MultiPixmapButton</class>
   <extends>QLabel</extends>
   <header>widget/multipixmapbutton.h</header>
   <slots>
    <signal>currentIndexChanged(int)</signal>
   </slots>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../icons.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>tcpConversationsView</sender>
   <signal>currentIndexChanged(QModelIndex)</signal>
   <receiver>HttpppMainWindow</receiver>
   <slot>forwardSelection(QModelIndex)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>85</x>
     <y>239</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>246</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tcpPacketsView</sender>
   <signal>currentIndexChanged(QModelIndex)</signal>
   <receiver>HttpppMainWindow</receiver>
   <slot>forwardSelection(QModelIndex)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>306</x>
     <y>308</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>314</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushTcpConversations</sender>
   <signal>toggled(bool)</signal>
   <receiver>HttpppMainWindow</receiver>
   <slot>changePanelVisibility(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>423</x>
     <y>49</y>
    </hint>
    <hint type="destinationlabel">
     <x>467</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushTcpPackets</sender>
   <signal>toggled(bool)</signal>
   <receiver>HttpppMainWindow</receiver>
   <slot>changePanelVisibility(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>504</x>
     <y>49</y>
    </hint>
    <hint type="destinationlabel">
     <x>561</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushHttpHits</sender>
   <signal>toggled(bool)</signal>
   <receiver>HttpppMainWindow</receiver>
   <slot>changePanelVisibility(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>585</x>
     <y>49</y>
    </hint>
    <hint type="destinationlabel">
     <x>651</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushDetails</sender>
   <signal>toggled(bool)</signal>
   <receiver>HttpppMainWindow</receiver>
   <slot>changePanelVisibility(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>666</x>
     <y>49</y>
    </hint>
    <hint type="destinationlabel">
     <x>731</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushLoad</sender>
   <signal>clicked()</signal>
   <receiver>HttpppMainWindow</receiver>
   <slot>loadFileDialog()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>52</x>
     <y>38</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>50</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushHitsToCSV</sender>
   <signal>clicked()</signal>
   <receiver>HttpppMainWindow</receiver>
   <slot>httpHitsToCsvDialog()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>231</x>
     <y>39</y>
    </hint>
    <hint type="destinationlabel">
     <x>222</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>httpHitsView</sender>
   <signal>currentIndexChanged(QModelIndex)</signal>
   <receiver>HttpppMainWindow</receiver>
   <slot>forwardSelection(QModelIndex)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>578</x>
     <y>308</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>367</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushMessages</sender>
   <signal>toggled(bool)</signal>
   <receiver>HttpppMainWindow</receiver>
   <slot>changePanelVisibility(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>722</x>
     <y>38</y>
    </hint>
    <hint type="destinationlabel">
     <x>751</x>
     <y>24</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushClearMessages</sender>
   <signal>clicked()</signal>
   <receiver>messages</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>736</x>
     <y>401</y>
    </hint>
    <hint type="destinationlabel">
     <x>585</x>
     <y>376</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushReload</sender>
   <signal>clicked()</signal>
   <receiver>HttpppMainWindow</receiver>
   <slot>reload()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>143</x>
     <y>41</y>
    </hint>
    <hint type="destinationlabel">
     <x>100</x>
     <y>22</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushCloseMessages</sender>
   <signal>clicked()</signal>
   <receiver>pushMessages</receiver>
   <slot>toggle()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>692</x>
     <y>361</y>
    </hint>
    <hint type="destinationlabel">
     <x>705</x>
     <y>39</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>changePanelVisibility(bool)</slot>
  <slot>forwardSelection(QModelIndex)</slot>
  <slot>loadFileDialog()</slot>
  <slot>startCaptureDialog()</slot>
  <slot>httpHitsToCsvDialog()</slot>
  <slot>reload()</slot>
 </slots>
</ui>
